src/net/tcp/tcp.cpp

```diff
@@ -131,7 +131,7 @@ seq_t TCP::generate_iss() {
 */
 port_t TCP::next_free_port() {
 
-  current_ephemeral_ = (current_ephemeral_ == 0) ? current_ephemeral_ + 1025 : current_ephemeral_ + 1;
+  current_ephemeral_ = (current_ephemeral_ == port_ranges::DYNAMIC_END) ? port_ranges::DYNAMIC_START : current_ephemeral_ + 1;
 
   // Avoid giving a port that is bound to a service.
   while(listeners_.find(current_ephemeral_) != listeners_.end())
```

Comment:
I think the below line (`current_ephemeral_++;`) contains a bug. The value of `current_ephemeral_` may overflow the `port_ranges::DYNAMIC_END` if the last port from the dynamic port range is used by the listener. Am I right?